server/server.go

```diff
@@ -348,17 +350,21 @@ func newServer(cfg Config, metrics *Metrics) (*Server, error) {
 		WithRequest:              !cfg.ExcludeRequestInLog,
 		DisableRequestSuccessLog: cfg.DisableRequestSuccessLog,
 	}
+	reportGRPCStatusesOption := middleware.DoNotAcceptGRPCStatusesOption
+	if cfg.ReportGRPCCodesInInstrumentationLabel {
+		reportGRPCStatusesOption = middleware.AcceptGRPCStatusesOption
+	}
 	grpcMiddleware := []grpc.UnaryServerInterceptor{
```

Comment:
We should pass the option only to ENABLE the new reporting. If old behaviour is wanted, let's not pass any option. That way we only need one, while preserving compatibility with existing code.